lib/DataStore.js

```diff
@@ -311,6 +311,45 @@ DataStore.prototype.getMappingsForChannelByOrigin = function(server, channel, or
     });
 };
 
+DataStore.prototype.getModesForChannel = function (server, channel) {
+    log.info("getModesForChannel (server=%s, channel=%s)",
+        server.domain, channel
+    );
+    let remoteId = IrcRoom.createId(server, channel);
+    return this._roomStore.getEntriesByRemoteId(remoteId).then((entries) => {
+        const mapping = {};
+        entries.forEach((entry) => {
+            mapping[entry.matrix.getId()] = entry.remote.get("modes") || [];
+        });
+        return mapping;
+    });
+};
+
+DataStore.prototype.setModeForRoom = Promise.coroutine(function*(roomId, mode, enabled=True) {
+    log.info("setModeForRoom (mode=%s, roomId=%s, enabled=%s)",
+        mode, roomId, enabled
+    );
+    this._roomStore.getEntriesByMatrixId(roomId).then((entries) => {
+        entries.forEach((entry) => {
+            const modes = entry.remote.get("modes") || [];
+            const hasMode = modes.includes(mode);
+
+            if ((!hasMode && !enabled) || (hasMode && enabled)) {
```

Comment:
A comment on this would be great, given it took me a while to figure out what is going on. Also, this could be shortened to `hasMode === enabled` if you wanted.

Reply:
...yes, it could couldn't it.